internal/csi-addons/networkfence/fencing.go

```diff
@@ -286,6 +298,113 @@ func (nf *NetworkFence) AddClientEviction(ctx context.Context) error {
 	return nil
 }
 
+func parseBlocklistEntries(output string) []string {
+	// Parse the stdout to extract blocklist entries
+	// variable entries is a slice, contain the output of above command as separate strings
+	// it will remove any empty lines and the 1st line which is "listed x entries"
+	// example, entries[0] = "127.0.0.1:0/3710147553 2018-03-19 11:32:24.716146"
+	lines := strings.Split(output, "\n")
+	var entries []string
+	for _, line := range lines {
+		if strings.TrimSpace(line) != "" && !strings.HasPrefix(line, "listed") {
+			entries = append(entries, line)
+		}
+	}
+
+	return entries
+}
+
+func (nf *NetworkFence) getBlocklistEntries(ctx context.Context) ([]string, error) {
+	arg := []string{
+		"--id", nf.cr.ID,
+		"--keyfile=" + nf.cr.KeyFile,
+		"-m", nf.Monitors,
+	}
+	// FIXME: replace the ceph command with go-ceph API in future
+	cmd := []string{"osd", "blocklist", "ls"}
+	// $ ceph osd blocklist ls
+	// '''listed 1 entries
+	// 127.0.0.1:0/3710147553 2018-03-19 11:32:24.716146'''
+	cmd = append(cmd, arg...)
+	stdout, stdErr, err := util.ExecCommandWithTimeout(ctx, 5*time.Minute, "ceph", cmd...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list blocklisted clients: %w, stderr: %q", err, stdErr)
+	}
+
+	entries := parseBlocklistEntries(stdout)
+
+	return entries, nil
+}
+
+func (nf *NetworkFence) removeBlocklistEntry(ctx context.Context, entry string) error {
+	arg := []string{
+		"--id", nf.cr.ID,
+		"--keyfile=" + nf.cr.KeyFile,
+		"-m", nf.Monitors,
+	}
+	// FIXME: replace the ceph command with go-ceph API in future
+	cmd := []string{"osd", "blocklist", "rm", entry}
+	cmd = append(cmd, arg...)
+	_, stdErr, err := util.ExecCommandWithTimeout(ctx, 5*time.Minute, "ceph", cmd...)
+	if err != nil {
+		return fmt.Errorf("failed to un-blocklist entry %s: %w, stderr: %q", entry, err, stdErr)
+	}
+	log.DebugLog(ctx, "un-blocklisted entry %s successfully", entry)
+
+	return nil
+}
+
+func fetchIPfromEntry(entry string) (string, error) {
+	// example, entry = "127.0.0.1:0/3710147553 2018-03-19 11:32:24.716146"
+	// then returning value will be 127.0.0.1
+	if parts := strings.Fields(entry); len(parts) >= 2 {
+		ip := strings.Split(parts[0], ":")[0]
+
+		return ip, nil
+	}
+
+	return "", errors.New("failed to extract IP address")
+}
+
+// RemoveClientEviction unblocks access for all the IPs in the IP range mentioned via the CIDR block
+// using a network fence.
+func (nf *NetworkFence) RemoveClientEviction(ctx context.Context) error {
+	// get the list of blocklist entries
+	blocklistEntries, err := nf.getBlocklistEntries(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get blocklist entries: %w", err)
+	}
+
+	// iterate through CIDR blocks and remove matching entries
+	for _, cidr := range nf.Cidr {
+		for _, entry := range blocklistEntries {
+			// fetch IP from entry
+			entryIP, err := fetchIPfromEntry(entry)
+			if err != nil {
+				return fmt.Errorf("error fetching client IP: %w", err)
```

Comment:
Consider including the `entry` contents in the error message (with `%q`). It'll make troubleshooting and debugging much easier.